### PR TITLE
fix: automate update template dependencies as part of lerna publish

### PR DESCRIPTION
**Pull request by raymondfeng**
The PR allows template deps to be updated as part of lerna publish.

1. Move `templateDependencies` into `packages/cli/package.json` as `config.templateDependencies`
2. `update-template-deps` will be run at `loopback-next` level for the `version` phase (after all package.json files have been updated.)
3. `packages/cli/package.json` will be refreshed 
4. lerna will commit the change as part of publish

This will prevent issues like https://github.com/strongloop/loopback-next/issues/1166.

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "bootstrap": "npm i && lerna bootstrap",
     "release": "npm run build:full && lerna publish",
     "update-template-deps": "node bin/update-template-deps -f",
+    "version": "npm run update-template-deps",
```
Comment (virkt25) [inline package.json]: What triggers the `version` script?

Comment (raymondfeng) [reply package.json]: `lerna publish` will trigger the `version` scripts as follows:

1. Run `preversion` at root
2. Run `preversion` for each package
3. Update package.json and CHANGELOG for each package
4. Run 'version` for each package
5. Run `version` at root

Comment (raymondfeng) [reply package.json]: It follows https://docs.npmjs.com/cli/version

File: packages/example-rpc-server/package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "tslint": "lb-tslint",
     "tslint:fix": "npm run tslint -- --fix",
     "pretest": "npm run clean && npm run build",
-    "test": "lb-dist mocha --opts node_modules/@loopback/build/mocha.opts DIST/test",
+    "test": "lb-mocha DIST/test",
```
Comment (virkt25) [inline packages/example-rpc-server/package.json]: I think it's better to make unrelated changes in a different PR. (for future).

**Comment by raymondfeng** [comment]
> I was trying this out by changing some deps in templateDependencies and dry running lerna publish (npx lerna publish --skip-git --skip-npm) but I didn't see any output/changes from this script. Was I doing it wrong?

I just tried as follows:
```
git checkout automate-update-template-deps
npx lerna publish --skip-git --skip-npm
```

Now see changes in `packages/cli/package.json` with `config.templateDependenecies` being updated with the newer versions.

**Pull request merged**